xmtp_mls/src/identity_updates.rs

```diff
@@ -75,6 +78,33 @@ where
 
         Ok(get_state(updates)?)
     }
+
+    pub fn get_association_state_diff<InboxId: AsRef<str>>(
+        &self,
+        conn: &'a DbConnection<'a>,
+        inbox_id: String,
+        starting_sequence_id: Option<i64>,
+        ending_sequence_id: Option<i64>,
+    ) -> Result<AssociationStateDiff, ClientError> {
+        let initial_state = self.get_association_state(conn, &inbox_id, starting_sequence_id)?;
+        if starting_sequence_id.is_none() {
+            return Ok(AssociationStateDiff::from(initial_state));
```

Comment:
I find this semantically confusing to `From` a state into a state diff. Does it make sense inline the logic here and remove `From`?
```
AssociationStateDiff {
            new_members: state.members.keys().cloned().collect(),
            removed_members: vec![],
        }
```

Reply:
Another option could be to reverse the operands in `initial_state.diff(&final_state)` to `final-state.diff(&initial_state)`, and allow `initial_state` to be null or empty

Reply:
Ya, it is a bit of a stretch. If it's confusing I can just inline it or make a separate function